src/services.rs

```diff
@@ -551,71 +551,112 @@ pub async fn create_or_update_service(
+                for trait_ in t {
+                    if trait_.trait_type == "daemonscaler" {
+                        return Some(trait_);
+                    }
+                }
+            };
+            None
+        })
+        .collect();
 
-                // Store the set of labels for this component so that we can match them to hosts
-                // when creating the label selector on the service.
-                if let Some(traits) = &component.traits {
-                    for t in traits {
-                        // The only way we know how to properly create a service without
-                        // being told to in the manifest is if we're using a daemonscaler.
-                        // That guarantees a k8s service can route traffic to any pod in a
-                        // deployment and that it will actually handle the inbound request.
-                        // Alternatively we could try spying on wadm commands and
-                        // reconciling host inventories, but that might not be worth it.
-                        if t.trait_type != "daemonscaler" {
-                            continue;
-                        }
-                        if let TraitProperty::SpreadScaler(scaler) = &t.properties {
-                            for spread in scaler.spread.iter() {
-                                spread.requirements.iter().for_each(|(k, v)| {
-                                    details
-                                        .labels
-                                        .get_or_insert_with(HashMap::new)
-                                        .insert(k.clone(), v.clone());
-                                });
-                            }
-                        }
-                        return Some(details);
+    debug!(scalers=?scalers, "Found scalers");
+    // Right now we only support daemonscalers, so if we don't find any then we have nothing to do
+    if scalers.is_empty() {
+        return None;
+    }
+
+    let links: Vec<&Trait> = components
+        .iter()
+        .filter_map(|c| {
+            if let Some(t) = &c.traits {
+                for trait_ in t {
+                    if trait_.trait_type == "link" {
+                        return Some(trait_);
+                    }
+                }
+            };
+            None
+        })
+        .collect();
+
+    let mut details = HttpServerComponent::default();
+    let mut should_create_service = false;
+    for l in links {
+        if let TraitProperty::Link(props) = &l.properties {
+            if props.namespace == "wasi"
+                && props.package == "http"
+                && props.interfaces.contains(&"incoming-handler".to_string())
+            {
+                should_create_service = true;
+                for p in props.source_config.iter() {
+                    if let Some(config_props) = &p.properties {
+                        if let Some(addr) = config_props.get("ADDRESS") {
```

Comment:
I'm curious why is the ADDRESS capitalized here? Looking at the provider code it didn't seem like that defines it as capitalized, but all lowercase, am I missing some conversion logic that happens in between?